src/components/componentContainer.ts

```diff
@@ -6,54 +6,36 @@ module Plottable {
    * It will not do anything if instantiated directly.
    */
   export class ComponentContainer extends Component {
-    private _components: Component[] = [];
-
     public anchor(selection: D3.Selection) {
       super.anchor(selection);
-      this.components().forEach((c) => c.anchor(this._content));
+      this._components().forEach((c) => c.anchor(this._content));
       return this;
     }
 
     public render() {
-      this._components.forEach((c) => c.render());
+      this._components().forEach((c) => c.render());
       return this;
     }
 
     /**
-     * Removes the specified Component from the ComponentContainer
-     * 
-     * @param c Component the Component to remove.
+     * Removes the specified Component from the ComponentContainer.
      */
-    public remove(c: Component) {
-      var removeIndex = this._components.indexOf(c);
-      if (removeIndex >= 0) {
-        this.components().splice(removeIndex, 1);
+    public remove(component: Component) {
```

Comment:
The signatures of the two `add()` methods are different, which is why `add()` doesn't exist at the `ComponentContainer` level.